src/common/ceph_mutex.h

```diff
@@ -77,7 +77,7 @@ namespace ceph {
 namespace ceph {
   typedef ceph::mutex_debug mutex;
   typedef ceph::mutex_recursive_debug recursive_mutex;
-  typedef ceph::condition_variable_debug condition_variable;
+  typedef ceph::condition_variable_debug<ceph::mutex_debug> condition_variable;
```

Comment:
instead of templatizing `condition_variable_debug`. how about just using `std::condition_variable_any` when we need to pass a `fair_mutex` to `condition_variable`? so we can still have the lockdep checks offered by `ceph::mutex` in debug mode?

Reply:
Let me try this.

Reply:
I tried it and it works.

Let me add it back in later PRs if needed when supporting the lockdep debug feature for fair mutex. Currently for the mds_lock the lockdep feature makes no sense.

Reply:
`condition_variable_debug` is supposed to be an implementation of `std::condition_variable`. it offers the sanity checks and lockdeps checks.

> Currently for the mds_lock the lockdep feature makes no sense.

i am afraid to support lockdep features was not the point of my comment. i was commenting on your efforts to add the debugging facilities to `fair_mutex`, as it seemed that you were trying to implement them.

i think, `std::condition_variable_any` is designed to to support different mutex types, not `std::condition_variable` or its ceph implementation.

Reply:
This works for mds code. 

I am also planing to support fair mutex in client code in late future if needed. Which the lockdep will make much sense with the inode lock PR in https://github.com/ceph/ceph/pull/39797.

Let me do that in later PRs when needed.